compiler/rustc_mir/src/borrow_check/diagnostics/region_errors.rs

```diff
@@ -332,6 +333,18 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
             }
         };
 
+        match variance_info {
+            ty::VarianceDiagInfo::None => {}
+            ty::VarianceDiagInfo::Mut { ty } => {
+                diag.note(&format!(
+                    "requirement occurs because of a mutable reference/pointer to {}",
+                    ty
+                ));
+                diag.note("mutable references/pointers are invariant over their type parameter");
+                diag.help("See <https://doc.rust-lang.org/nomicon/subtyping.html> for more information about variance");
```

Comment:
```suggestion
                diag.help("see <https://doc.rust-lang.org/nomicon/subtyping.html> for more information about variance");
```

Reply:
Actually, it might be even better to join this help with the note above. The length is not ideal, but we can rely on the user's cli reflow for text.

Reply:
I think it makes sense to have this message as a distinct 'footer', which gets applied to all variance-related messages.